### PR TITLE
fix: don't add swapper multiple times

### PR DESCRIPTION
**Pull request by 0xdef1cafe**
## Description

**hooks are not singletons**

* move the instantiation of the swapper manager to a singleton and setup the swappers there
* also addresses bug where we were incorrectly setting up avalanche zrx swapper as an eth mainnet swapper
* flags adding avalanche zrx swapper behind existing avalanche flag

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

Closes https://github.com/shapeshift/web/issues/2123

## Risk

swapper regressions

## Testing

have a play setting up trades and watch the console for errors
unlikely needed to actually execute trades

## Screenshots (if applicable)

n/a

### CODE REVIEW

File: src/components/Trade/hooks/useSwapper/useSwapper.ts [hunk 1]
```diff
@@ -56,51 +112,7 @@ export const useSwapper = () => {
...
-    })
-
-    try {
-      swapperManager.addSwapper(zrxEthereumSwapper)
-      swapperManager.addSwapper(zrxAvalancheSwapper)
-    } catch (e) {
-      moduleLogger.error(e, { fn: 'addSwapper' }, 'error adding swapper')
-    }
-  }, [adapterManager, swapperManager])
+  const [swapperManager] = useState<SwapperManager>(getSwapperManager())
```
Comment (gomesalexandre) [inline src/components/Trade/hooks/useSwapper/useSwapper.ts:115]: Feels weird only getting the state field and not the setter but I get the idea on how we want to be reactive to the `Maybe<SwapperManager>` unwrapping, and I don't really see any other way to achieve this as long as this is a hook so 🍬

**Review by gomesalexandre — DISMISSED** [review]
Checked it out locally against develop, and this fixes the `error adding swapper` errors 🍬

**Review by 0xApotheosis — APPROVED** [review]
Looks good once tests fixed.

File: src/components/Trade/hooks/useSwapper/useSwapper.ts (hunk 1 above)
Comment (0xdef1cafe) [reply src/components/Trade/hooks/useSwapper/useSwapper.ts:115]: @elmutt and i gave this a try yeah, using `getSwapperManager()` everywhere works and has the same effect, but stylistically looks odd.

it's not actually a `Maybe` - it's concrete but w/ side effects

**Review by elmutt — APPROVED** [review]

**Review by 0xApotheosis — APPROVED** [review]
🍬

**Pull request merged**